server/route/AppointmentController.js

```diff
@@ -53,16 +276,22 @@ router.post(
 			);
 		}
 
-		// Create a new appointment
+		// Create a new appointment with references
 		const newAppointment = new Appointment({
-			clientUsername,
-			stylistUsername,
+			client: client._id,
+			stylist: stylist._id,
 			appointmentDate,
 			duration,
 			notes,
 		});
 
 		const savedAppointment = await newAppointment.save();
+
+		// Populate client and stylist details before sending the response
+		await savedAppointment
+			.populate("client", "info.name")
+			.populate("stylist", "info.name");
```

Comment:
When I test create appointment using Postman, I get the following error:
<img width="806" alt="Screen Shot 2024-11-07 at 12 34 27 PM" src="https://github.com/user-attachments/assets/511fc524-0a17-43f0-a81e-e6f40ed8818c">